server/BraveAlerterConfigurator.js

```diff
@@ -81,28 +81,44 @@ class BraveAlerterConfigurator {
 
   async alertSessionChangedCallback(alertSession) {
     if (alertSession.alertState === undefined && alertSession.incidentCategoryKey === undefined) {
-      return
+      return {}
     }
 
     let pgClient
     let session
+    let replaceReturnMessageToRespondedByPhoneNumber
+    let replaceReturnMessageToOtherResponderPhoneNumbers
 
     try {
       pgClient = await db.beginTransaction()
       if (pgClient === null) {
         helpers.logError(`alertSessionChangedCallback: Error starting transaction`)
-        return
+        return {}
       }
       session = await db.getSessionWithSessionId(alertSession.sessionId, pgClient)
 
       if (session) {
+        helpers.log(`Current session state: ${session.chatbotState}; alert session state: ${alertSession.alertState}`)
+
         // If this is not a OneSignal session (i.e. we are given a respondedByPhoneNumber) and the session has no respondedByPhoneNumber, then this is the first SMS response, so assign it as the session's respondedByPhoneNumber
         if (alertSession.respondedByPhoneNumber !== undefined && session.respondedByPhoneNumber === null) {
           session.respondedByPhoneNumber = alertSession.respondedByPhoneNumber
         }
 
         // If this is a OneSignal session (i.e. it isn't given the respondedByPhoneNumber) or if the SMS came from the session's respondedByPhoneNumber
         if (alertSession.respondedByPhoneNumber === undefined || alertSession.respondedByPhoneNumber === session.respondedByPhoneNumber) {
+          if (
+            alertSession.alertState === CHATBOT_STATE.RESET &&
+            session.numberOfAlerts < helpers.getEnvVar('SESSION_MIN_NUMBER_OF_ALERTS_TO_ALLOW_RESET_REQUEST')
+          ) {
```

Comment:
Do you mind explaining why you added `session.numberOfAlerts < helpers.getEnvVar('SESSION_MIN_NUMBER_OF_ALERTS_TO_ALLOW_RESET_REQUEST')` to this conditional? 

Are we allowing the sensor to be reset if the chatbot state is RESET and the number of alerts is more than our threshold? My understanding is that once it hits the RESET state that this ends the session (so it shouldn't matter how many alerts there are) but I think I'm misunderstanding. Thanks!

Reply:
Never replied to this :( ; but to explain, this environment variable was added to define the threshold at which the RESET alert state is valid; it is checked against to determine if a reset request is valid or invalid.

This PR operates around using `alertSession.alertState` as a __proposed__ state, rather than a __final__ state. It comes from the alert state machine and whatever message the client sent. It becomes RESET when the client says "reset", and then alertSessionChangedCallback function checks the number of alerts in this session before it updates the database entry for this session to become RESET. This if statement is where it determines a reset request as valid or invalid.